### PR TITLE
Fix appian join tests

### PR DESCRIPTION
**Pull request by Tmonster**
This should fix the otherside of https://github.com/duckdblabs/duckdb-internal/issues/2611

There were some regressions caused exclusively by the join order optimizer. This PR fixes those regressions, and also makes a benchmark from the appian queries. 

The idea behind this PR is to prefer right deep trees so that not as many hash tables occupy memory at the same time.

@maiadegraaf I also had to fix some adbc tests, don't know if that is something you wanna take a look at?

### CODE REVIEW

**Comment by Tishj** [comment]
ADBC was worked on by @pdet and myself, I believe Pedro created those join tests

File: benchmark/appian_benchmarks/appian.benchmark.in [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# name: ${FILE_PATH}
+# description: ${DESCRIPTION}
+# group: [clickbench]
```
Comment (lnkuiper) [inline benchmark/appian_benchmarks/appian.benchmark.in:3]: `[appian_benchmarks]`

File: .github/regression/appian.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+benchmark/appian_benchmarks/q01.benchmark
+benchmark/appian_benchmarks/q02.benchmark
+benchmark/appian_benchmarks/q03.benchmark
+benchmark/appian_benchmarks/q04.benchmark
+benchmark/appian_benchmarks/q05.benchmark
+benchmark/appian_benchmarks/q06.benchmark
+benchmark/appian_benchmarks/q07.benchmark
+benchmark/appian_benchmarks/q08.benchmark
```
Comment (lnkuiper) [inline .github/regression/appian.csv:8]: Does this need to be added to Regression.yml so CI runs it? If not, how does the performance compare with `main`?

**Review by lnkuiper — COMMENTED** [review]
Thanks for the PR! LGTM, just two comments:

**Comment by Mytherin** [comment]
Thanks! Looks great

**Pull request merged**